<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <node pkg="lidar_align" name="lidar_align" type="lidar_align" launch-prefix="xterm -hold -e ">
    <param name="input_bag_path" value="/home/z/Downloads/extrinsics.bag" />
    <param name="input_csv_path" value="/home/z/Downloads/calib0_opt_poses.csv" />
    <param name="output_pointcloud_path" value="$(find lidar_align)/results/$(anon lidar_points).ply" />
    <param name="output_calibration_path" value="$(find lidar_align)/results/$(anon calibration).txt" />

    <param name="use_n_scans" value="1000" />
    <param name="local" value="false" />
    <param name="transforms_from_csv" value="true"/>
    <param name="time_cal" value="true" />
    <param name="motion_compensation" value="false" />
    <param name="clockwise_lidar" value="false" />
    <rosparam param="inital_guess">[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]</rosparam>
    <rosparam param="range">[0.5, 0.5, 0.5, 3.2, 3.2, 3.2, 3.2]</rosparam>
    <param name="max_evals" value="2000" />
    <param name="xtol" value="0.0000001" />
    
    <param name="min_point_distance" value="2" />
    <param name="max_point_distance" value="10" />
    <param name="keep_points_ratio" value="0.005" />
    <param name="max_per_scan_points" value="10000" />

    <param name="knn_batch_size" value="5000" />
    <param name="knn_k" value="1" />
    <param name="knn_max_dist" value="1.0" />
  </node>

</launch>